백다연/MvvmStudy/app/src/main/java/com/test/mvvmstudy/viewmodel/SearchViewModel.kt

```diff
@@ -1,26 +1,33 @@
 package com.test.mvvmstudy.viewmodel
 
-import android.util.Log
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.test.mvvmstudy.data.SearchResultDetail
+import androidx.lifecycle.viewModelScope
+import com.test.mvvmstudy.data.NetworkResult
 import com.test.mvvmstudy.repository.SearchRepository
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
 class SearchViewModel : ViewModel() {
 
     private val repository = SearchRepository()
 
-    private val _result = MutableLiveData<List<SearchResultDetail>>()
-    val resultList: LiveData<List<SearchResultDetail>>
-        get() = _result
-
-    val isLoading = MutableLiveData<Boolean>()
+    private val _resultStateFlow: MutableStateFlow<NetworkResult> =
+        MutableStateFlow(NetworkResult.Empty)
+    val resultStateFlow: StateFlow<NetworkResult>
+        get() = _resultStateFlow
 
     fun getSearchData(query: String) {
-        repository.searchRepository(query,
-            onSuccess = { result -> _result.value = result },
-            onFailure = { error -> Log.d("error", error) })
+        viewModelScope.launch {
+            _resultStateFlow.value = NetworkResult.Loading
+            repository.searchRepository(query)
+                .catch { e ->
+                    _resultStateFlow.value = NetworkResult.ErrorDataResult(exception = e)
```

Comment:
줄임말보다는 풀어서 설명해줘도 좋을 것 같아요!
```suggestion
                .catch { exception ->
                    _resultStateFlow.value = NetworkResult.ErrorDataResult(exception = exception)
```